invenio_accounts/models.py

```diff
@@ -90,6 +121,44 @@ class User(db.Model, Timestamp, UserMixin):
         "LoginInformation", back_populates="user", uselist=False, lazy="joined"
     )
 
+    @hybrid_property
+    def user_profile(self):
+        """Get the user profile."""
+        # NOTE: accessing this property requires an initialized app for config
+        if self._user_profile is None:
+            return None
+        elif not isinstance(self._user_profile, UserProfileDict):
+            self._user_profile = UserProfileDict(**self._user_profile)
```

Comment:
it has to be seen if this setting of the column value causes issues down the road (currently thinking of the post-commit hooks in users-resources, since you're absolutely not allowed to do anything that causes SQL statements to be issued after a session's been committed)
i could imagine that this change doesn't actually cause any net changes, and should thus be fine
otherwise, we might have to take the (probably minimal) performance hits and rewrite this line to always convert the stored dict to a `UserProfileDict` rather than caching it:
```python
return UserProfileDict(**self._user_profile)
```

Reply:
i've reworked this line to be on the safer side, as i don't think that the performance hit is too dramatic here (the user profiles shouldn't be large enough to make copying & validating the dictionaries have a big impact, and then it only happens when the user's profile property is accessed)